visits-service-new/src/main/java/com/petclinic/visits/visitsservicenew/BusinessLayer/VisitServiceImpl.java

```diff
@@ -80,11 +88,20 @@ public Mono<VisitResponseDTO> addVisit(Mono<VisitRequestDTO> visitRequestDTOMono
                         .then(validatePetId(visitRequestDTO.getPetId()))
                         .then(validateVetId(visitRequestDTO.getPractitionerId()))
                         .then(Mono.just(visitRequestDTO))
+                        .doOnNext(s->{
+
+                            authServiceClient.getUserById(visitRequestDTO.getJwtToken(), visitRequestDTO.getOwnerId()).subscribe(user->mailService.sendMail(generateVisitRequestEmail(user, visitRequestDTO.getPetId(), visitRequestDTO.getVisitDate())));
```

Comment:
can you explain what .subscribe () is doing here?

Reply:
It allows to run code once we get the return from the asynchronous getUserById method from the AuthServiceClient. In this case, it is a non blocking way to wait to get the user, then go ahead with sending the email.